cluster/aws/util.sh

```diff
@@ -435,6 +435,44 @@ function ensure-master-pd {
   fi
 }
 
+# Configures a CloudWatch alarm to reboot the instance on failure
+function reboot-on-failure {
+  local instance_id=$1
+
+  echo "Creating Cloudwatch alarm to reboot instance ${instance_id} on failure"
+
+  local aws_owner_id=`aws ec2 describe-instances --instance-ids ${instance_id} --query Reservations[0].OwnerId`
+  if [[ -z "${aws_owner_id}" ]]; then
+    echo "Unable to determinate AWS account id for ${instance_id}"
+    exit 1
+  fi
+
+  aws cloudwatch put-metric-alarm \
+                 --alarm-name k8s-${instance_id}-statuscheckfailure-reboot \
+                 --alarm-description "Reboot ${instance_id} on status check failure" \
+                 --namespace "AWS/EC2" \
+                 --dimensions Name=InstanceId,Value=${instance_id} \
+                 --statistic Minimum \
+                 --metric-name StatusCheckFailed \
+                 --comparison-operator GreaterThanThreshold \
+                 --threshold 0 \
+                 --period 60 \
+                 --evaluation-periods 3 \
```

Comment:
My understanding is that threshold=0 indicates when the condition is breached.  StatusCheck is a binary condition, 0 when healthy and 1 when unhealthy.  Evaluation Periods indicates the number of consecutive breaches before the alarm triggers.  So I think this means that this will reboot only after 3 minutes of continuous (statistic=Minimum) status check = 0.

e.g. "Recover an Amazon Elastic Compute Cloud instance" here http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/quickref-cloudwatch.html

I agree the docs are less than clear.   But I did try this (by logging in and taking eth0 down) and it did seem to work as expected.

One note: this is just software failure on the instance.  I am not yet enabling the ec2:recover action for failure of the host, because that has a whole list of conditions and is also very difficult to test.  Not sure if I should do that too (in a separate PR)?